### PR TITLE
Use SES for sending emails

### PR DESCRIPTION
**Pull request by jonathaningram**

### CODE REVIEW

File: admin/app/notify_client/complaint_api_client.py [hunk 1]
```diff
@@ -0,0 +1,18 @@
+from app.notify_client import NotifyAdminAPIClient
+
+
+class ComplaintApiClient(NotifyAdminAPIClient):
+    # Fudge assert in the super __init__ so
+    # we can set those variables later.
+    def __init__(self):
+        super().__init__("a" * 73, "b")
```
Comment (dalehurwitz) [inline admin/app/notify_client/complaint_api_client.py]: Im curious, what are the mentioned variables?

File: api/app/notifications/notifications_ses_callback.py [hunk 2]
```diff
@@ -1,105 +1,193 @@
...
-    client_name = 'SES'
-    try:
-        errors = validate_callback_data(data=ses_request, fields=['Message'], client_name=client_name)
-        if errors:
-            return errors
+register_errors(ses_callback_blueprint)
+
+
+@ses_callback_blueprint.route('/notifications/email/ses', methods=['POST'])
+def sns_callback_handler():
```
Comment (dalehurwitz) [inline api/app/notifications/notifications_ses_callback.py]: What's sns?

**Comment by jonathaningram** [comment]
AWS Simple Notification Service.

**Comment by jonathaningram** [comment]
The API key and secret passed to the client, I believe.

File: admin/app/templates/views/callbacks.html [hunk 3]
```diff
@@ -53,9 +53,9 @@ <h2 class="heading-medium">Text messages you receive</h2>
         caption_visible=False
       ) %}
         {% for key, description, format in [
-          ('id', 'Notify’s id for the received message', 'UUID'),
+          ('id', 'Notify’s ID for the received message', 'UUID'),
           ('source_number', 'The phone number the message was sent from', '447700912345'),
```
Comment (andrewoh531) [inline admin/app/templates/views/callbacks.html]: Change country code of example?

File: api/app/__init__.py [hunk 4]
```diff
@@ -146,6 +155,8 @@ def register_blueprint(application):
     application.register_blueprint(status_blueprint)
 
     # delivery receipts
+    ses_callback_blueprint.before_request(requires_no_auth)
```
Comment (andrewoh531) [inline api/app/__init__.py]: Unless I missed it, we should verify messages sent to this endpoint are from AWS to protect ourselves from DOS attacks. It's unlikely to happen now when we're still early stages but when use picks up we should consider implementing it.

https://docs.aws.amazon.com/sns/latest/dg/SendMessageToHttp.verify.signature.html

File: .circleci/config.yml [hunk 5]
```diff
@@ -29,6 +29,17 @@ references:
...
       at: .
 
+  aws_install: &aws_install
+    run:
+      command: |
+        curl "https://s3.amazonaws.com/aws-cli/awscli-bundle.zip" -o "awscli-bundle.zip"
+        unzip awscli-bundle.zip
+        ./awscli-bundle/install -b ~/bin/aws
+        ls -la ~/.local/lib/
+        ls -la $HOME/bin/
```
[the remark this answers is not in the document]
Comment (jonathaningram) [reply .circleci/config.yml]: Was removed, your comment was probably from an older version ⚡️

File: admin/app/templates/views/callbacks.html (hunk 3 above)
Comment (jonathaningram) [reply admin/app/templates/views/callbacks.html]: Done thanks.

File: api/app/__init__.py (hunk 4 above)
Comment (jonathaningram) [reply api/app/__init__.py]: You're right. This link will give us some existing Python implementations as it doesn't look like there's any official SDK support: https://github.com/boto/boto3/issues/1469

**Pull request closed without merging**